src/chunk.c

```diff
@@ -1099,21 +1099,47 @@ chunk_create_from_hypercube_and_table_after_lock(const Hypertable *ht, Hypercube
 	chunk->hypertable_relid = ht->main_table_relid;
 	Assert(OidIsValid(ht->main_table_relid));
 
+	Relation chunk_rel = table_open(chunk_table_relid, AccessExclusiveLock);
+	Relation ht_rel = table_open(ht->main_table_relid, AccessExclusiveLock);
+	TupleDesc tupdesc = RelationGetDescr(chunk_rel);
+
+	for (int attno = 0; attno < tupdesc->natts; attno++)
```

Comment:
If possible, we should probably move this validation up before creating the chunk object above. This will make sure we are not incrementing the chunk_id unnecessarily. It will also reduce the changes in the test.